### ISSUE TITLE
Unable to pipe between 2 ipfs commands

### ISSUE THREAD

**Issue by dylanPowers** [ipfs/go-ipfs#1394]
Specifically I'm unable to run `ipfs cat <hash> | ipfs add` without getting the error:
```
Error: Post http://127.0.0.1:5001/api/v0/add?encoding=json&progress=true&stream-channels=true: dial tcp 127.0.0.1:5001: connection refused
```

FYI, the daemon is offline

**Comment by whyrusleeping** [comment]
What i beleive is happening here is that the first command takes the repo lock, and the second command, seeing that the lock is taken, assumes that a daemon is online and tries to connect to it.

**Comment by jbenet** [comment]
> the second command, seeing that the lock is taken, assumes that a daemon is online and tries to connect to it.

yeah this is why i separated `repo.lock` and `api` in https://github.com/ipfs/specs/tree/master/repo/fs-repo#api -- the fsrepo doesnt look like that. and that spec needs to be updated with Tv's pinning stuff.

**Comment by Kubuxu** [comment]
This bug is back, now running command like that with daemon disabled returns:
```
Error: resource temporarily unavailable
```

**Comment by Stebalien** [comment]
Proposed solution: Always use a daemon (but start an ephemeral daemon on-demand).

Timeline: someday :(
